gips/0083.md

```diff
@@ -0,0 +1,250 @@
+
+Here is a sample payload:
+
+```json
+{
+  "type": "substreamsServiceRegistration",
+  "data": {
+    "endpoint": "https://mynetwork.substreams.example.com",
+    "service": "substreams-v1",
+    "network": "mynetwork",
+    "providerLogo": "https://company.example.com/logo.png",
+    "providerName": "My Company",
+    "providerId": "my-company",
+  }
+}
+```
+
+Here is a sample revocation:
+
+```json
+{
+  "type": "substreamsServiceRevocation",
+  "data": {
+    "service": "substreams-v1",
+    "network": "mynetwork",
+    "providerId": "my-company",
+  }
+}
+```
+
+The triplet `service`, `network` and `providerId` is what uniquely identifies a Substreams service published on the network.
+
+The service `substreams-v1` means that the endpoint makes available a gRPC endpoint responding to these methods:
+
+- sf.substreams.rpc.v2.Stream/Block
+- sf.substreams.rpc.v2.EndpointInfo/Info
+
+as specified by https://github.com/streamingfast/substreams/blob/develop/proto/sf/substreams/rpc/v2/service.proto updated from time-to-time in backwards compatible ways.
+
+The Payment Gateway MUST listen to the Arbitrum DataEdge contract, for registrations and revocation (through a has a Substreams module streamed, a Subgraph or other means), and update its local view of the network.
+
+The Payment Gateway CAN have systems to perform health checks (e.g. on `/health` or `/info` endpoints), or other checks to ensure active, up-to-date and properly configured providers are offered to end users. A Payment Gateway CAN check block height to ensure the backing provider is close to chain head before offering it to users.
+
+
+### Tools
+
+THIS SECTION IS NON-NORMATIVE. Different tools can exist to do those operations. We are listing some for informational purposes here.
+
+For Indexers, in order to register a new Substreams provider on-chain, and show up on the Payment Gateway, you can use commands such as:
+
+```
+substreams network register --operator-wallet 0x123123123 --operator-priv-key-file my.key --service substreams-v1 --endpoint mainnet.eth.example.com --provider-logo https://www.example.com/static/mylogo.png --provider-id example-company --provider-name "Example Company" --network ethereum
+```
+
+This command allows providers to register their service, specifying their operator wallet, private key file, the service type (`substreams-v1`), the endpoint URL, logo, name, and the supported network. The registration data is signed by the operator's key.  An update to an existing registration can be achieved by re-running the command with updated parameters, based on the `service`, `providerId` and `network` tripled.  A corresponding `unregister` or `revoke` command would be added to allow providers to take down their service from the front-end.
+
+### References
+
+ - [DataEdge addresses](https://github.com/graphprotocol/contracts/blob/main/packages/data-edge/addresses.js)
```

Comment:
This link is broken, I think you meant addresses.json? But in any case, I think it would be better to deploy a new instance of the DataEdge, note its address down in addresses.json, but also mention the address explicitly either in the GIP or in the GGP when the Council votes for this GIP.

Reply:
ok, fixing link, and adding notes in a new "DataEdge address" section. I think we can ask the Council to choose either a new deployment, or an existing one.. and we can put the address in the vote (or update this document then).

Reply:
ok, we're using Substreams Service Deployments instead now.. so becomes moot.